internal/bminventory/inventory.go

```diff
@@ -5234,29 +5234,25 @@ func (b *bareMetalInventory) V2GetHost(ctx context.Context, params installer.V2G
 }
 
 func (b *bareMetalInventory) V2UpdateHostInstallProgress(ctx context.Context, params installer.V2UpdateHostInstallProgressParams) middleware.Responder {
-	h, err := b.V2UpdateHostInstallProgressInternal(ctx, params)
+	err := b.V2UpdateHostInstallProgressInternal(ctx, params)
 	if err != nil {
-		if h == nil {
-			return installer.NewV2UpdateHostInstallProgressNotFound().
-				WithPayload(common.GenerateError(http.StatusNotFound, err))
-		}
-		return installer.NewV2UpdateHostInstallProgressInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		return common.GenerateErrorResponder(err)
 	}
 	return installer.NewV2UpdateHostInstallProgressOK()
 }
-func (b *bareMetalInventory) V2UpdateHostInstallProgressInternal(ctx context.Context, params installer.V2UpdateHostInstallProgressParams) (*common.Host, error) {
+func (b *bareMetalInventory) V2UpdateHostInstallProgressInternal(ctx context.Context, params installer.V2UpdateHostInstallProgressParams) error {
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("Update host %s install progress", params.HostID)
 	host, err := common.GetHostFromDB(b.db, params.InfraEnvID.String(), params.HostID.String())
 	if err != nil {
 		log.WithError(err).Errorf("failed to find host %s", params.HostID)
-		return nil, err
+		return common.NewApiError(http.StatusNotFound, err)
```

Comment:
I don't think this is definitely a 404. For example if the DB happens to be down or if we have some network error this should be 500. Since we use `GenerateErrorResponder` in the caller this can just return the error directly.

Reply:
Thanks for catching all these! Modified it!